model-engine/model_engine_server/domain/use_cases/llm_model_endpoint_use_cases.py

```diff
@@ -1365,6 +1365,11 @@ def validate_and_update_completion_params(
                 "include_stop_str_in_output is only supported in vllm."
             )
 
+    if (
+        request.guided_choice or request.guided_regex or request.guided_json
+    ) and not inference_framework == LLMInferenceFramework.VLLM:
```

Comment:
also check that only 1 of the following is provided?

Reply:
sure